src/objectid.ts

```diff
@@ -42,10 +33,17 @@ function makeObjectIdError(invalidString, index) {
   );
 }
 
+export interface ObjectIdLike {
```

Comment:
shouldn't this be:
```
export type ObjectIdLike = ObjectId | {
  id: string | Buffer;
  toHexString(): string;
}
```

Reply:
looks like this comment was lost in the page refresh

Reply:
Sorry about that again, I don't see the benefit to unioning the two if something accepts ObjectIdLike things it will accept ObjecId as well since it satisfies the minimal shape of the ObjectIdLike interface. Meanwhile if something accepts `ObjectIdLike | ObjectId` things you can have runtime errors if you don't check that the additional properties on ObjectId exist on the thing you're operating on. 
For example PromiseLike is only a .then method and doesn't include anything more. Also our DBRefLike isn't a union of the actual DBRef becuase it is supposed to be an object that lacks the toExtendedJSON method as an example.